cdap-app-templates/cdap-etl/cdap-etl-api-spark/src/main/java/io/cdap/cdap/etl/api/batch/SparkCompute.java

```diff
@@ -22,17 +22,40 @@
 import io.cdap.cdap.etl.api.SubmitterLifecycle;
 import java.io.Serializable;
 import org.apache.spark.api.java.JavaRDD;
+import org.apache.spark.sql.Dataset;
+import org.apache.spark.sql.Row;
 
 /**
  * Spark Compute stage.
  *
- * @param <IN> Type of input object
- * @param <OUT> Type of output object
+ * @param <I> Type of input object
+ * @param <O> Type of output object
  */
 @Beta
-public abstract class SparkCompute<IN, OUT>
+public abstract class SparkCompute<I, O>
     implements SubmitterLifecycle<SparkPluginContext>, PipelineConfigurable, Serializable {
 
+  /**
+   * List of modes that plugin prefers to operate in.
+   *
+   * @see #getMode(SparkExecutionPluginContext)
+   */
+  public enum Mode {
```

Comment:
there is a similar use case for SparkSink plugins, so it would be good to pull this out of the SparkCompute class so it could be re-used there in the future

Reply:
Done